src/org/zaproxy/zap/extension/httpsinfo/ExtensionHttpsInfo.java

```diff
@@ -93,20 +84,34 @@ public void hook(ExtensionHook extensionHook) {
 
 		if (getView() != null) {
 			extensionHook.getHookMenu().addPopupMenuItem(getPopupMsgMenu());
+			extensionHook.getHookView().addStatusPanel(getHttpsInfoPanel());
 		}
-
 	}
 
 	private MenuEntry getPopupMsgMenu() {
-		if (httpsEntry == null) {
-			httpsEntry = new MenuEntry(this.getMessageString("httpsinfo.rightclick.menuitem"), this);
-			httpsEntry.setIcon(new ImageIcon(ExtensionHttpsInfo.class.getResource(ICON_PATH)));
+		if (httpsMenuEntry == null) {
+			httpsMenuEntry = new MenuEntry(Constant.messages.getString("httpsinfo.rightclick.menuitem"), this);
+			httpsMenuEntry.setIcon(new ImageIcon(ExtensionHttpsInfo.class.getResource(ICON_PATH)));
 		}
-		return httpsEntry;
+		return httpsMenuEntry;
 	}
-
-	public String getMessageString(String key) {
-		return Constant.messages.getString(key);
+	
+	protected TabbedPanel2 getHttpsInfoTabsPanel() {
+		if (httpsInfoTabsPanel == null) {
+			httpsInfoTabsPanel = new TabbedPanel2();
```

Comment:
This tabbed pane does not remove the tabs when closed just hides them, currently it's not possible to remove "old" analysis.

Reply:
Ya, that's why I decided to append a number.

Reply:
I thought that was just to differentiate the different queries (not because we couldn't remove them). Be good if it allowed to do that (also worth noting that the icon of the tab changes after hiding/showing it).

Reply:
Well a bit of both I guess. I’ll look at creating a tabbedpanel3 or something ;)

Reply:
Sounds good to me.

Reply:
Regarding the UI tweaks, I can take a look and guide with the changes required but need to know how do you want the tabbed panel to behave. I think we just need to change it to remove the tabs instead of hide them?

Reply:
I'm off tomorrow and the following Friday as well, I'll try to catch you in irc to discuss. 

At a high level yes I think that's mainly the goal.

Reply:
OK.